### DISCUSSION TITLE
v2 routes - draft

### DISCUSSION THREAD

**Discussion by arxdsilva** [prest/prest#854] [API Server]
# Current routes

<img width="767" alt="image" src="https://github.com/prest/prest/assets/9372503/9a398b81-8d9e-4519-bbb3-d64272cf1681">

# problems

1. no URL versioning (unclear on which version it was released)
2. When using it as a framework, any route with `/a/b` will be sent to `/{database}/{schema}`, same happens with CRUD routes

# proposal

1. add versioning to URL, such as: `/v2/query/view/{db}/{schema}/{view}`, `/v2.2/query/_QUERIES/{location}/{script}`
2. separate URLs by query domain, such as: `views`, `query`, `plugin`, and `custom`, and place them in the path structures

New routes:

| Endpoint | operation |
| --- | --- |
| `/v2/_health` | healthcheck |
| `/v2/db/{database}/auth` | db-specific authentication |
| `/v2/views/databases` | available dbs in the db server |
| `/v2/views/schemas` | available schemas in the db server |
| `/v2/views/tables` | available tables in the db server |
| `/v2/views/{db}/{schema}` | get schema struct |
| `/v2.1/views/{db}/{schema}/{table}` | get table struct |
| `/v2.3/plugin/{file}/{func}` | execute plugin |
| `/v2/custom/{location}/{script}` | CRUD ops on script |
| `/v2.5/query/{db}/{schema}/{table}` | CRUD operations |
| `/v2/query/batch/{db}/{schema}/{table}` | POST with batch insert |

**Comment by avelino** [comment]
cc @fabriziomello @vmesel @felipeweb

**Reply by arxdsilva** [reply]
updated the proposed routes, hopefully it is easier to understand

**Comment by avelino** [comment]
@arxdsilva check this issue #435
